### ISSUE TITLE
CSP + DebugBar

### ISSUE THREAD

**Issue by nowackipawel** [codeigniter4/CodeIgniter4#1165]
Hi there, I'm not an CSP expert.
I configured CSP with self and required domains (for script style and fonts) and everything was gr8 until DebugToolbar was turned on . Even if all of toolbar's tabs  seams to work  ok... there are errors in console:

Content Security Policy: The page’s settings blocked the loading of a resource at self (“script-src”). Source: onclick attribute on A element. [only once]
Content Security Policy: The page’s settings blocked the loading of a resource at self (“style-src”). [repated when tab is changed]
Content Security Policy: The page’s settings blocked the loading of a resource at self (“script-src”). [repated when tab is changed]

[nginx/php7.2/debian + ff / iridium]

... actually iridium (chrome) gave me more details:
?debugbar:49 Refused to apply inline style because it violates the following Content Security Policy directive: "style-src 'self' https://maxcdn.bootstrapcdn.com/ https://use.fontawesome.com/ 'nonce-fd68498a9d2a9ea28cd45f26'". Either the 'unsafe-inline' keyword, a hash ('sha256-47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU='), or a nonce ('nonce-...') is required to enable inline execution.

xhttp.onreadystatechange @ ?debugbar:49

I think problem is not with first html code which is generated by DebugToolbar but when it tried to apply additional scripts/styles.

**Comment by lonnieezell** [comment]
Are you using the latest version of the develop branch? I remember a patch came through not very long ago that set {csp_nonce} in the html for the toolbar which should have fixed that error.

**Comment by nowackipawel** [comment]
Unfortunately my version of  /system/Debug/Toolbar/toolbarloader.js.php and  /application/Filters/DebugToolbar.php are the same as here :(.

**Comment by puschie286** [comment]
your errors are "normal" and can be ignored because they doesnt effect on your site at all. the latest toolbar changes should only allow use with csp protection enabled and development environment.

**Comment by jim-parry** [comment]
Kint issue? out-of-scope for us? No further info in 3 months.

**Comment by crustamet** [comment]
I don`t know about you guys but I just created a new Codeigniter 4 project with the latest updates on PHP 8.1.2
And this problem still persist when I use development environment with the debugbar {csp-style-nonce} just not replacing in development mode or if it does replace it replace with empty.

![image](https://user-images.githubusercontent.com/36475250/204242958-b86be7f7-a892-4949-8b9a-3efa635e07a3.png)
![image](https://user-images.githubusercontent.com/36475250/204243046-d72be324-cc39-4d49-82d4-355d81de7836.png)

**Comment by crustamet** [comment]
I guess is hard to make a js file that does not use scripts inline and include them

**Comment by mavelo-llc** [comment]
> ![image](https://user-images.githubusercontent.com/36475250/204243046-d72be324-cc39-4d49-82d4-355d81de7836.png)

Even though it's not a critical flaw, would like to see this resolved. Inline styles are still being injected by debugbar as of 4.6.3

Looks like the nonce is not being applied before `toolbar.innerHTML = responseText` within `xhttp.onreadystatechange` of /vendor/codeigniter4/framework/system/Debug/Toolbar/Views/toolbarloader.js

**Comment by michalsn** [comment]
https://codeigniter.com/user_guide/outgoing/csp.html#turning-csp-on

**Comment by mavelo-llc** [comment]
Turned on and functional, everywhere except where mentioned.

**Comment by mavelo-llc** [comment]
CSP nonces all work as expected, except here...

<img width="1213" height="467" alt="Image" src="https://github.com/user-attachments/assets/102dc72e-1330-47b1-8641-395bd3878994" />

> Looks like the nonce is not being applied before `toolbar.innerHTML = responseText` within `xhttp.onreadystatechange` of /vendor/codeigniter4/framework/system/Debug/Toolbar/Views/toolbarloader.js
